quickwit/quickwit-indexing/src/source/file_source.rs

```diff
@@ -140,114 +122,59 @@ impl TypedSourceFactory for FileSourceFactory {
         source_runtime: SourceRuntime,
         params: FileSourceParams,
     ) -> anyhow::Result<FileSource> {
-        let checkpoint = source_runtime.fetch_checkpoint().await?;
-        let mut offset = 0;
-
-        let reader: FileSourceReader = if let Some(filepath) = &params.filepath {
+        let file_source = if let Some(filepath) = &params.filepath {
```

Comment:
nitpick... 

We tend to prefer early exit style in our codebase.

`let Some else` makes it quite natural here.

```rust
let Some(filepath) = &params.filepath else {
    return Ok(FileSource {
                source_id: source_runtime.source_id().to_string(),
                reader: DocFileReader::from_stdin(),
                partition_id: None,
                counters: FileSourceCounters::default(),
     });
};
let partition_id = PartitionId::from(filepath.to_string_lossy().borrow());
let checkpoint = source_runtime.fetch_checkpoint().await?;
...
 Ok(reader)